File: app/src/main/java/gm/tieba/tabswitch/hooker/OpenSign.java
package gm.tieba.tabswitch.hooker;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import java.util.Calendar;

import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.XposedHelpers;
import gm.tieba.tabswitch.BaseHooker;
import gm.tieba.tabswitch.IHooker;
import gm.tieba.tabswitch.dao.Preferences;

public class OpenSign extends BaseHooker implements IHooker {
    public void hook() throws Throwable {
        XposedHelpers.findAndHookMethod("com.baidu.tieba.tblauncher.MainTabActivity", sClassLoader, "onCreate", Bundle.class, new XC_MethodHook() {
            @Override
            protected void afterHookedMethod(MethodHookParam param) throws Throwable {
                Activity activity = (Activity) param.thisObject;
                if (!Preferences.getIsSigned() && Calendar.getInstance().get(Calendar.HOUR_OF_DAY) != 0) {
                    Intent intent = new Intent().setClassName(activity, "com.baidu.tieba.signall.SignAllForumActivity");
                    activity.startActivity(intent);
                }
            }
        });
        XposedHelpers.findAndHookMethod("com.baidu.tieba.signall.SignAllForumActivity", sClassLoader, "onClick", View.class, new XC_MethodHook() {
            @Override
            protected void afterHookedMethod(MethodHookParam param) throws Throwable {
                Activity activity = (Activity) param.thisObject;
                if (!Preferences.getIsSigned() && Calendar.getInstance().get(Calendar.HOUR_OF_DAY) != 0) {
                    Preferences.putSignDate();
                    activity.finish();
                }
            }
        });
    }
}

File: app/src/main/java/gm/tieba/tabswitch/hooker/auto/OpenSign.kt
package gm.tieba.tabswitch.hooker.auto

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import de.robv.android.xposed.XC_MethodHook
import de.robv.android.xposed.XposedHelpers
import gm.tieba.tabswitch.XposedContext
import gm.tieba.tabswitch.dao.Preferences.getIsSigned
import gm.tieba.tabswitch.dao.Preferences.putSignDate
import gm.tieba.tabswitch.hooker.IHooker
import java.util.Calendar

class OpenSign : XposedContext(), IHooker {
    override fun key(): String {
        return "open_sign"
    }

    @Throws(Throwable::class)
    override fun hook() {
        hookAfterMethod(
            "com.baidu.tieba.tblauncher.MainTabActivity",
            "onCreate", Bundle::class.java
        ) { param ->
            val activity = param.thisObject as Activity
            if (!getIsSigned() && Calendar.getInstance()[Calendar.HOUR_OF_DAY] != 0) {
                val intent = Intent().setClassName(activity, "com.baidu.tieba.signall.SignAllForumActivity")
                activity.startActivity(intent)
            }
        }
        hookAfterMethod(
            "com.baidu.tieba.signall.SignAllForumActivity",
            "onClick", View::class.java
        ) { param ->
            val activity = param.thisObject as Activity
            if (!getIsSigned() && Calendar.getInstance()[Calendar.HOUR_OF_DAY] != 0) {
                putSignDate()
                activity.finish()
            }
        }
    }
}
